crypto/bls.go

```diff
@@ -421,28 +421,25 @@ func hashToG1(data []byte) *pointG1 {
 }
 
 // This is only a TEST function.
-// It wraps a call to optimized SwU algorithm since cgo can't be used
-// in go test files
-func OpSwUUnitTest(output []byte, input []byte) {
-	C.opswu_test((*C.uchar)(&output[0]),
-		(*C.uchar)(&input[0]),
-		(C.int)(len(input)))
-}
-
-// This is only a TEST function.
-// It wraps a call to signing using Relic to get a curve point, internally using XMD:SHA256
+// signWithXMDSHA256 signs a message using XMD_SHA256 as a hash to field.
 //
-// (since cgo can't be used in go test files)
-func (sk *PrKeyBLSBLS12381) signWithRelicMapTest(data []byte) Signature {
-	var point pointG1
-	mapToG1RelicTest(&point, data, []byte("BLS_SIG_BLS12381G1_XMD:SHA-256_SSWU_RO_NUL_"))
-
-	// set BLS context
-	blsInstance.reInit()
-
+// The function is in this file because cgo can't be used in go test files.
+// TODO: implement a hasher for XMD SHA256 and use the `Sign` function.
+func (sk *PrKeyBLSBLS12381) signWithXMDSHA256(data []byte) Signature {
```

Comment:
This should no longer be needed, as we can make `blst` sign with our own hasher for comparison (see `useHash` flag [here](https://github.com/supranational/blst/blob/eb6151961c133a930420e844e1a84708fbb4f6a4/bindings/go/blst.go#L1904-L1922))

Reply:
Ah this is great!! I was wondering if I could avoid this! Thanks for pointing it out

Reply:
It looks like `useHash` in BLST doesn't seem to do what we thought it was doing :
- if `useHash = true`, we enter a `HashToG2` [function](https://github.com/supranational/blst/blob/eb6151961c133a930420e844e1a84708fbb4f6a4/bindings/go/blst.go#L288) and end up [here](https://github.com/supranational/blst/blob/eb6151961c133a930420e844e1a84708fbb4f6a4/src/map_to_g2.c#L388).
- if `useHash = false`, we enter a `EncodeToG2` [function](https://github.com/supranational/blst/blob/eb6151961c133a930420e844e1a84708fbb4f6a4/bindings/go/blst.go#L290) and end up [here](https://github.com/supranational/blst/blob/eb6151961c133a930420e844e1a84708fbb4f6a4/src/map_to_g2.c#L373).

Unlike what the function names say, both functions are actually hashing the message and dst together first. The difference is that one hashes to 2 field elements and the other hashes to one field element only. 
Both functions use the simplified SWU map but with [different](https://github.com/supranational/blst/blob/eb6151961c133a930420e844e1a84708fbb4f6a4/src/map_to_g2.c#L355) constructions (map from field to curve twice and add the results Vs map from field to curve once)

Reply:
Yeah, that does seem to be an XMD SHA256 in both cases, and either the [uniform or non-uniform encoding](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-11#section-3). in other words, we're looking at `BLS12381G1_XMD:SHA-256_SSWU_NU_` (hash to curve) or `BLS12381G1_XMD:SHA-
   256_SSWU_RO_` (encode to curve).

Reply:
Yes, exactly